json_merger/merger.py

```diff
@@ -261,7 +262,8 @@ def _recursive_merge(self, root, head, update, key_path=()):
                 )
                 new_obj = self._recursive_merge(root_obj, head_obj, update_obj,
                                                 absolute_key_path + (idx, ))
-                new_list.append(new_obj)
+                if new_obj != NOTHING:
+                    new_list.append(new_obj)
```

Comment:
could you add a more realistic test, that also tests this feature? you can use the author with removed ORCID in HEAD.